src/Stratis.Bitcoin/Controllers/NodeController.cs

```diff
@@ -358,8 +358,10 @@ public async Task<IActionResult> GetTxOutAsync([FromQuery] string trxid, uint vo
         [HttpPost]
         [Route("shutdown")]
         [Route("stop")]
-        public IActionResult Shutdown()
+        public async Task<IActionResult> ShutdownAsync([FromBody] int delayInSeconds = 0)
         {
+            await Task.Delay(TimeSpan.FromSeconds(delayInSeconds)).ConfigureAwait(false);
```

Comment:
Why would you change the method to be async just to insert a delay?

Reply:
Or is it somehow required by `[FromBody]`?

Reply:
I did that because I was looking for a parameter that would not be completely useless.
I could have added a boolean "areYouSure" but it is stupid. @bokobza  and I agreed this was adding a little value, as well as making the method safe

Reply:
you could always go with Thread.Sleep though Without changing the method to async. That is what I'm talking about here. I will need to check what you are talking about hopefully later today

Reply:
If the user wants to delay the shutdown for 30 min, Thread.Sleep might be a problem.

Reply:
You don't have cancellation token so you will have problem even with task delay, if you want to remove that problem, you need to have cancellation token node lifetime